### PR TITLE
AEEE-25086, AEEE-25087, AEEE-25652, AEEE-25714, AEEE-25805, AEEE-25806, AEEE-25808, AEEE-25095: Create Event implementation for filling all fields

### PR DESCRIPTION
**Pull request by yashjain1394**
The following tasks have been implemented for filling all fields in create event flows: 

1. Create Event-Basic Info page implementation
2. Create Event-Speakers & Host page implementation
3. Create Event-Additional content page implementation
4. Create Event-RSVP page implementation

Additionally, The following feedbacks have been implemented:

1. Implement search on the ECC dashboard after event creation
2. Move page name validation to common methods
3. Create separate feature files for all fields and required fields
4. Parameterize name for event creation
5. Reporting for the create event flow
6. Moved productnametovalue and removeSpacefromcategory to the Page Object.
7. Updated the feature scenario to "Create events with all fields and validate."
8. Clearly added the "Signed-in" step in the feature file.
9. Segregated basic info and other fields.
10. StartDate set to 26, EndDate set to 28.
11. Implemented Playwright's Built-in Handle approach to handle nested shadow DOM elements.
12. Updated the feature file name to "createEventRequiredFields" & "createEventAllFields".
13. Done refactoring to make the script more robust & reduce flakiness.
14. Separated the scenarios into "Search" and "Delete".

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,5 +2,8 @@
 #Run below command to run the test case @registerEvent
 npx run src/e2e -b webkit -p stage -t "@registerEvent"
 
-#Run below command to run the test case @createEvent
-npx run src/e2e -b webkit -p stage -t "@createEvent"
+#Run below command to run the test case @createEventWithRequiredFields
```
Comment (adgaurgit) [inline README.md]: can we have short crisp name of the feature

File: src/e2e/features/createEventWithAllFields.feature [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  Scenario: Verify ECC dashboard page content
+    Given I am on the ECC dashboard page in signed-out state
+    Then I should see the All Events label on the page
+    Then I should see the Search box on the page
+    Then I should see the table headers on the page
+    Then I should see pagination container
+    Then I should see the footer section on the page
+
+    Scenario Outline: Validate and create events with all fields
+    Given I am on the ECC dashboard page in signed-in state
```
Comment (adgaurgit) [inline src/e2e/features/createEventWithAllFields.feature]: where are we signing in?

File: src/e2e/features/createEventWithAllFields.feature [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+  Scenario: Verify ECC dashboard page content
+    Given I am on the ECC dashboard page in signed-out state
+    Then I should see the All Events label on the page
+    Then I should see the Search box on the page
+    Then I should see the table headers on the page
+    Then I should see pagination container
+    Then I should see the footer section on the page
+
+    Scenario Outline: Validate and create events with all fields
```
Comment (adgaurgit) [inline src/e2e/features/createEventWithAllFields.feature]: scneario name should be in sync with steps, we are first creating and then validating

File: src/e2e/features/createEventWithAllFields.feature [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    Given I am on the ECC dashboard page in signed-in state
+    And I should see the Create new event button on the page
+    Then I should be able to click on Create new event button
+    Then I am in the create event flow and Basic info page
+    Then I fill out create event Basic info page with <EventDataWithAllFields> and click Next step
+    Then I fill out create event Speakers & Hosts page with <EventDataWithAllFields> and click Next step
+    Then I fill out create event Additional content page with <EventDataWithAllFields> and click Next step
+    Then I fill out create event RSVP page with <EventDataWithAllFields> and click Publish event
+    Then I should check that event is created
+    Then I should be able to search & validate the event on ECC dashboard
```
Comment (adgaurgit) [inline src/e2e/features/createEventWithAllFields.feature]: should search and delete be another scenario?

File: src/e2e/features/createEventWithRequiredFields.feature
Comment (adgaurgit) [inline src/e2e/features/createEventWithRequiredFields.feature:1]: same comments as on all fields file

File: src/e2e/pages/createEvent_basicInfo.page.js
Comment (adgaurgit) [inline src/e2e/pages/createEvent_basicInfo.page.js:1]: lets review this together

**Review by adgaurgit — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (yashjain1394) [reply README.md]: As discussed, I have changed the name to createEventRequiredFields & createEventAllFields

File: src/e2e/features/createEventWithAllFields.feature (hunk 3 above)
Comment (yashjain1394) [reply src/e2e/features/createEventWithAllFields.feature]: I have updated it to "Create event with all fields and validate"

File: src/e2e/features/createEventWithAllFields.feature (hunk 4 above)
Comment (yashjain1394) [reply src/e2e/features/createEventWithAllFields.feature]: There are some challenges to this. We can discuss this offline.

File: src/e2e/features/createEventWithRequiredFields.feature
Comment (yashjain1394) [reply src/e2e/features/createEventWithRequiredFields.feature:1]: Updated the feature file "createEventRequiredFields"

File: src/e2e/pages/createEvent_basicInfo.page.js
Comment (yashjain1394) [reply src/e2e/pages/createEvent_basicInfo.page.js:1]: We have reviewed this offline, and I have implemented the feedback.

File: src/e2e/features/createEventWithAllFields.feature (hunk 2 above)
Comment (yashjain1394) [reply src/e2e/features/createEventWithAllFields.feature]: I have added additional step for sign-in

File: src/e2e/features/createEventWithAllFields.feature (hunk 4 above)
Comment (yashjain1394) [reply src/e2e/features/createEventWithAllFields.feature]: Separated Search & delete scenarios

**Comment by adgaurgit** [comment]
Please add additional feedback in next PR

**Pull request merged**